src/vm/regions.rs

```diff
@@ -40,5 +40,21 @@ pub mod regions_view_model {
 
             regions_vm
         }
+
+        pub fn from_enabled_ids(slot:& SaveSlot, enabled_ids: &Vec<u32>) -> Self {
+            let mut regions_vm = RegionsViewModel::default();
+
+            for i in 0..enabled_ids.len() {
+                let key = &slot.regions.unlocked_regions[i as usize];
+                let is_invadeable_region = ID_TO_REGION.lock().unwrap().contains_key(key);
+                
+                if is_invadeable_region {
+                    let region = ID_TO_REGION.lock().unwrap()[key];
+                    regions_vm.regions.get_mut(&region).expect("").0 = true;
```

Comment:
Maybe you can use `unwrap()` instead of `expect("")` if no error message is needed.

Aslo maybe split it into two lines to give the `.0` a self descriptive name
```rust
let (activated, _,_,_) = regions_vm.regions.get_mut(&region).unwrap();
*activated = true;
```

Reply:
I think I should reserve this refactor until later, so we can focus on functionality in this PR.